src/twisted/test/test_udp.py

```diff
@@ -307,6 +307,66 @@ def cbFinished(ignored):
         d.addCallback(cbFinished)
         return d
 
+    def test_server_read_failure(self):
+        """
+        When a server fails to successfully read a packet the server should
+        still be able to process future packets.
+        The IOCP reactor had a historical problem where a failure to read caused
+        the reactor to ignore any future reads. This test should prevent a regression.
+
+        Note: This test assumes no one is listening on port 80 UDP.
```

Comment:
This should allocate a port in the safe way for tests (i.e.: listen on port zero and get the bound port).

Reply:
So, I believe in this test I actually am correctly using the "listen on zero" when trying to write to the actual port.
However, I see how my comment about port 80 is confusing. Port 80 in this test actually represents a port that intentionally has no listener. So I don't actually want the port to be allocated and the test never does try to allocate on it.

The reason I need a guarantee that some port is not listening is so that we can trigger windows to respond immediately to the recvfrom and trigger the buggy situation.

I agree that just assuming nobody is listening on port 80 UDP is probably not ideal though. Do you have any ideas on how we might be able to figure out a port that we can guarantee has no listener?

Reply:
I can't think of an official way to reserve a port without listening on it, and your approach seems reasonably reliable to me (using a port in the reserved range, but which has no defined use).

I had the same immediate misunderstanding as @glyph, so it might be worth amending the `Note:` comment to make it clearer. I think it's fine as it is, though.


```suggestion
        Note: This test assumes no one is listening on port 80 UDP, so that we
        can be sure to elicit a failure when we write.
```